src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixCUDA.java

```diff
@@ -2937,6 +2937,113 @@ public static void solve(ExecutionContext ec, GPUContext gCtx, String instName,
 
 		//debugPrintMatrix(b, n, 1);
     }
+	
+	/**
+	 * Implements the "svd" function
+	 *
+	 * @param ec         a valid {@link ExecutionContext}
+	 * @param gCtx       a valid {@link GPUContext}
+	 * @param instName   the invoking instruction's name for record {@link Statistics}.
+	 * @param in         input matrix A
+	 * @param out1       output matrix U
+	 * @param out2       output matrix S
+	 * @param out3       output matrix Vt
+	 * @throws DMLRuntimeException if an error occurs
+	 */
+	public static void solve(ExecutionContext ec, GPUContext gCtx, String instName, MatrixObject in, MatrixObject out1, MatrixObject out2, MatrixObject out3 ) throws DMLRuntimeException {
```

Comment:
Name this `svd`.
The `out1`, `out2`, `out3` are not used. The output is not being set.